src/preamp.jl

```diff
@@ -25,18 +25,40 @@ This struct is currently mutable because in the case of noise modelling based on
     τ_rise::typeof(1.0*ns_unit) = T(15)*u"ns"
 
     "Preamp offset in keV"
-    offset::typeof(1.0*energy_unit) = 0u"keV";
+    offset_keV::typeof(1.0*energy_unit) = 0u"keV";
 
-    "Maximum DAQ energy"
-    max_e::typeof(1.0*energy_unit) = 10_000u"keV" # == typemax(UInt16)
+    "Preamp offset in ADC (alternative)"
+    offset_ADC = 0; # ADC units
+
+    # "Maximum DAQ energy"
+    # max_e::typeof(1.0*energy_unit) = 0u"keV" # == typemax(UInt16)
 
     "Preamp gain"
     # If offset = 0, it means gain has to be inferred (from the data baseline offset)
     # How does it end up being Float64?
-    gain = offset == 0u"keV" ? 0 : typemax(UInt16) / ((max_e+offset)/uconvert(u"keV", germanium_ionization_energy))
+    # gain =             
+    #     if offset_keV == 0u"keV"
+    #         if offset_ADC == 0
+    #             0
+    #         else
+    #             # ( typemax(UInt16) - offset_ADC ) * uconvert(u"keV", germanium_ionization_energy) / max_e # ADC/charge
+    #             ( typemax(UInt16) - offset_ADC ) / uconvert(u"eV", max_e) # ADC/eV
+    #         end
+    #     else
+    #         # typemax(UInt16) / ((max_e+offset_keV)/uconvert(u"keV", germanium_ionization_energy)) # ADC/charge
+    #         typemax(UInt16) / uconvert(u"eV", max_e + offset_keV) # ADC/eV
+    #     end            
+    gain::typeof(1.0/1u"eV")
```

Comment:
I think this one needs a default value.

Reply:
Well, it's different for every preamp, so the default value might be off. But you're right easier to have it for reference, same as tau decay and rise values. I will update them also to correspond more to reality, 15ns risetime is way too small

Reply:
What I mean is that the way this is implemented in code `@with_kw mutable struct GenericPreamp`
allowing for keyword arguments (`@with_kw`) works best if every field has a default value.
Of course, the default value might be off, but you can always overwrite it afterwards as it's a mutable struct.
(and I think you overwrite it anyhow most of the time).

How about a default value of `1.0u"eV^-1"`? Then, the pulse is given in units of eV.